amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/DefaultTableService.java

```diff
@@ -251,32 +255,94 @@ public Blocker block(
-        .buildBlocker();
+    Preconditions.checkNotNull(operations, "operations should not be null");
+    Preconditions.checkArgument(!operations.isEmpty(), "operations should not be empty");
+    Preconditions.checkArgument(blockerTimeout > 0, "blocker timeout must > 0");
+    String catalog = tableIdentifier.getCatalog();
+    String database = tableIdentifier.getDatabase();
+    String table = tableIdentifier.getTableName();
+    int tryCount = 0;
+    while (tryCount++ < 3) {
+      long now = System.currentTimeMillis();
+      doAs(
+          TableBlockerMapper.class,
+          mapper -> mapper.deleteExpiredBlockers(catalog, database, table, now));
+      List<TableBlocker> tableBlockers =
+          getAs(
+              TableBlockerMapper.class,
+              mapper ->
+                  mapper.selectBlockers(
+                      tableIdentifier.getCatalog(),
+                      tableIdentifier.getDatabase(),
+                      tableIdentifier.getTableName(),
+                      now));
+      if (TableBlocker.conflict(operations, tableBlockers)) {
+        throw new BlockerConflictException(operations + " is conflict with " + tableBlockers);
+      }
+      Optional<Long> maxBlockerOpt =
+          tableBlockers.stream()
+              .map(TableBlocker::getBlockerId)
+              .max(Comparator.comparingLong(l -> l));
+      long prevBlockerId = maxBlockerOpt.orElse(-1L);
+
+      TableBlocker tableBlocker =
+          TableBlocker.buildTableBlocker(
+              tableIdentifier, operations, properties, now, blockerTimeout, prevBlockerId);
+      try {
+        doAs(TableBlockerMapper.class, mapper -> mapper.insert(tableBlocker));
+        if (tableBlocker.getBlockerId() > 0) {
+          return tableBlocker.buildBlocker();
+        }
+      } catch (PersistenceException e) {
+        LOG.warn(
+            "Exception when create a blocker:{}, error message:{}", tableBlocker, e.getMessage());
+        if (e.getMessage() == null || !e.getMessage().contains("duplicate key")) {
+          LOG.error("Exception when create a blocker:{}", tableBlocker, e);
+        }
+      }
+    }
+    throw new BlockerConflictException("Failed to create a blocker: conflict meet max retry");
   }
 
   @Override
   public void releaseBlocker(TableIdentifier tableIdentifier, String blockerId) {
-    checkStarted();
-    TableRuntime tableRuntime = getRuntime(getServerTableIdentifier(tableIdentifier));
-    if (tableRuntime != null) {
-      tableRuntime.release(blockerId);
-    }
+    doAs(TableBlockerMapper.class, mapper -> mapper.deleteBlocker(Long.parseLong(blockerId)));
   }
 
   @Override
   public long renewBlocker(TableIdentifier tableIdentifier, String blockerId) {
-    checkStarted();
-    TableRuntime tableRuntime = getAndCheckExist(getServerTableIdentifier(tableIdentifier));
-    return tableRuntime.renew(blockerId, blockerTimeout);
+    int retry = 0;
+    while (retry++ < 3) {
```

Comment:
The same issue about using a literal value.

Reply:
fix